tests/client/core/test_middleware.js

```diff
@@ -0,0 +1,93 @@
+import { prefixMiddleWare } from 'core/middleware';
+
+describe('Prefix Middleware', () => {
+  let fakeRes;
+  let fakeNext;
+  let fakeConfig;
+
+  beforeEach(() => {
+    fakeNext = sinon.stub();
+    fakeRes = {
+      locals: {},
+      redirect: sinon.stub(),
+      set: sinon.stub(),
+    };
+    fakeConfig = new Map();
+    fakeConfig.set('validClientApplications', ['firefox', 'android']);
+  });
+
+  it('should call res.redirect', () => {
+    const fakeReq = {
+      originalUrl: '/en-us/firefox',
```

Comment:
Yeah, I should add more to this test to make it clearer or improve the description string.